### DISCUSSION TITLE
How to control gain size equally?

### DISCUSSION THREAD

**Discussion by kmmm70** [neperfepx/neper#233] [Q&A] [answered]
Hi All,

I used 
neper -T -n 100 -domain "cube(48,6,12)" -morpho "diameq:lognormal(1,0.5),1-sphericity:lognormal(0.145,0.03)" -reg 1
tried to make gain size, at least, as mush as equal to each others as shown in picture (they used Neper). My result is way far from those pictures. 

Please explain me about those number, 1, 0.5, 0.145, 0.03. What are they using for?

Thanks.
![2021-05-26_21-23-07](https://user-images.githubusercontent.com/57537961/119947852-67257200-bfc2-11eb-9e2f-87f36ec522a6.png)

![img1](https://user-images.githubusercontent.com/57537961/119947934-7efcf600-bfc2-11eb-9578-d0d9caf5782b.png)

**Comment by rquey** [comment, marked as answer]
This is described in the doc, Section A.1.4: the first number is the mean and the second the standard deviation.

You don't even have to use a distribution. If you want all grains of the same size:
```
neper -T -n 100 -domain "cube(48,6,12)" -morpho "diameq:1,1-sphericity:lognormal(0.145,0.03)" -reg 1
```

**Reply by kmmm70** [reply]
Awesome. Thanks a lot.

**Reply by kmmm70** [reply]
Romain,

When I run your command, I noticed that the processing is very slow, the number of iteration is more than xxx,xxx and keep going on. Sometimes I feel like it won't stop so I terminated it with "control+c". 
Is that something involving with openMP cause it was disabled during installation?
I would like to increase OMP_NUM_THREADS but it needs openMP installed, right? 

Thanks.

**Reply by rquey** [reply]
It will go much faster with multithreading. The CPU time is nearly inversely proportional to the number of threads up to about ~15 threads for this command.

To reduce the computation time, relax the convergence criterion. The default is `-morphooptistop 1e-6`. You can use a higher value.

**Reply by kmmm70** [reply]
Romain,

Thanks again.
